app/src/main/java/rkr/simplekeyboard/inputmethod/latin/RichInputConnection.java

```diff
@@ -560,6 +560,10 @@ public boolean hasCursorPosition() {
         return mExpectedSelStart != INVALID_CURSOR_POSITION && mExpectedSelEnd != INVALID_CURSOR_POSITION;
     }
 
+    public boolean lastCharacterIsSpace() {
```

Comment:
I think it would be better to move this out of `RichInputConnection` (which generically manages the input connection), and put it in `InputLogic` since this relates to a specific input handling case. The two methods that this calls are already public so I think this just adds clutter to `RichInputConnection`.